### PR TITLE
Flush firewall rules before setting it up

### PR DESCRIPTION
**Pull request by elyezer**
When before setting up any firewall rule make sure that no rule exists.

### CODE REVIEW

File: automation_tools/__init__.py [hunk 1]
```diff
@@ -382,6 +386,8 @@ def setup_firewall(definitions=None):
         if run('systemctl status firewalld', quiet=True).failed:
             run('systemctl enable firewalld')
             run('systemctl start firewalld')
+        if flush:
+            raise RuntimeWarning('flush for firewalld is not available')
```
Comment (elyezer) [inline automation_tools/__init__.py]: Tried to check how to do this but still have not found. Decided to send the PR this way to see if someone knows from the top of his/her head. Please do not merge before this line had changed to something that actually do the expected job.

**Comment by sthirugn** [comment]
If this PR will add value, can we get this merged for rhel6 and create an issue for rhel 7 please?

**Comment by elyezer** [comment]
@sthirugn it will avoid the cli.test_capsule failure we are currently having and will also ensure consistency between the rules we have on both systems.

**Comment by sthirugn** [comment]
Two questions:
* I think merging this code as it is not will not fail rhel 6 or rhel 7
* Can you please create a separate issue to fix rhel 7 so we can get this merged?

**Comment by elyezer** [comment]
@sthirugn I would like to merge this with the firewalld-cmd for cleaning up the rules, but I don't know much. Sent this in order to have other @SatelliteQE/quality-engineers folks looking over and helping me with it.

If we don't have any other reply until tomorrow end of the day then I will create an issue for providing the firewall-cmd and then we can merge this.

**Comment by rplevka** [comment]
@elyezer , perhaps this should do the trick?
```
[--permanent] --direct --remove-rules { ipv4 | ipv6 | eb } table chain
    Remove all rules in the chain with name chain exists in table
    table. This option concerns only rules previously added with
    --direct --add-rule in this chain.
```

**Comment by elyezer** [comment]
@rplevka that will remove rules added with the --direct option. I will try that later.

**Comment by elyezer** [comment]
Test run on RHEL7 with previous firewalld ports added (I manually forced the run of firewalld instead of iptables since the current code will use iptables for RHEL6 and 7):

```
fab setup_satellite_firewall -H
root@qe-sat6-rhel7.example.com -i ~/.ssh/id_jenkins_rsa
[root@qe-sat6-rhel7.example.com] Executing task
'setup_satellite_firewall'
rhel 7 2
[root@qe-sat6-rhel7.example.com] run: for port in
$(firewall-cmd --permanent --list-ports); do firewall-cmd --permanent
--remove-port="$port"; done
[root@qe-sat6-rhel7.example.com] out: success
[root@qe-sat6-rhel7.example.com] out: success
[root@qe-sat6-rhel7.example.com] out: success
[root@qe-sat6-rhel7.example.com] out: success
[root@qe-sat6-rhel7.example.com] out: success
[root@qe-sat6-rhel7.example.com] out: success
[root@qe-sat6-rhel7.example.com] out: success
[root@qe-sat6-rhel7.example.com] out: success
[root@qe-sat6-rhel7.example.com] out: success
[root@qe-sat6-rhel7.example.com] out: success
[root@qe-sat6-rhel7.example.com] out: success
[root@qe-sat6-rhel7.example.com] out: success
[root@qe-sat6-rhel7.example.com] out: success
[root@qe-sat6-rhel7.example.com] out: 

[root@qe-sat6-rhel7.example.com] run: for port in
$(firewall-cmd --list-ports); do firewall-cmd --remove-port="$port"; done
[root@qe-sat6-rhel7.example.com] run: firewall-cmd
--permanent --add-port="53/udp"
[root@qe-sat6-rhel7.example.com] out: success
[root@qe-sat6-rhel7.example.com] out: 

[root@qe-sat6-rhel7.example.com] run: firewall-cmd
--permanent --add-port="69/udp"
[root@qe-sat6-rhel7.example.com] out: success
[root@qe-sat6-rhel7.example.com] out: 

[root@qe-sat6-rhel7.example.com] run: firewall-cmd
--permanent --add-port="443/tcp"
[root@qe-sat6-rhel7.example.com] out: success
[root@qe-sat6-rhel7.example.com] out: 

[root@qe-sat6-rhel7.example.com] run: firewall-cmd
--permanent --add-port="5671/tcp"
[root@qe-sat6-rhel7.example.com] out: success
[root@qe-sat6-rhel7.example.com] out: 

[root@qe-sat6-rhel7.example.com] run: firewall-cmd
--permanent --add-port="80/tcp"
[root@qe-sat6-rhel7.example.com] out: success
[root@qe-sat6-rhel7.example.com] out: 

[root@qe-sat6-rhel7.example.com] run: firewall-cmd
--permanent --add-port="8140/tcp"
[root@qe-sat6-rhel7.example.com] out: success
[root@qe-sat6-rhel7.example.com] out: 

[root@qe-sat6-rhel7.example.com] run: firewall-cmd
--permanent --add-port="9090/tcp"
[root@qe-sat6-rhel7.example.com] out: success
[root@qe-sat6-rhel7.example.com] out: 

[root@qe-sat6-rhel7.example.com] run: firewall-cmd
--permanent --add-port="22/tcp"
[root@qe-sat6-rhel7.example.com] out: success
[root@qe-sat6-rhel7.example.com] out: 

[root@qe-sat6-rhel7.example.com] run: firewall-cmd
--permanent --add-port="5000/tcp"
[root@qe-sat6-rhel7.example.com] out: success
[root@qe-sat6-rhel7.example.com] out: 

[root@qe-sat6-rhel7.example.com] run: firewall-cmd
--permanent --add-port="5646/tcp"
[root@qe-sat6-rhel7.example.com] out: success
[root@qe-sat6-rhel7.example.com] out: 

[root@qe-sat6-rhel7.example.com] run: firewall-cmd
--permanent --add-port="5647/tcp"
[root@qe-sat6-rhel7.example.com] out: success
[root@qe-sat6-rhel7.example.com] out: 

[root@qe-sat6-rhel7.example.com] run: firewall-cmd
--permanent --add-port="8000/tcp"
[root@qe-sat6-rhel7.example.com] out: success
[root@qe-sat6-rhel7.example.com] out: 

[root@qe-sat6-rhel7.example.com] run: firewall-cmd
--permanent --add-port="8443/tcp"
[root@qe-sat6-rhel7.example.com] out: success
[root@qe-sat6-rhel7.example.com] out: 

rhel 7 2

Done.
Disconnecting from root@qe-sat6-rhel7.example.com... done.
```

File: automation_tools/__init__.py [hunk 2]
```diff
@@ -378,12 +380,23 @@ def setup_firewall(definitions=None):
             'iptables -I INPUT -m state --state NEW -p {0} --dport {1} '
             '-j ACCEPT'
         )
+        if flush:
```
Comment (sthirugn) [inline automation_tools/__init__.py]: Should the if statement in Line 377 have:

```if distro_info()[1] < 7:```

Comment (sthirugn) [inline automation_tools/__init__.py]: Never mind, currently we use iptables for both 6 and 7.

**Comment by sthirugn** [comment]
ACK

**Comment by rplevka** [comment]
ACK

**Pull request merged**